pkg/sql/codegen/proto/intermediate_representation.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+
+package proto;
+
+message FieldMeta {
+    string name = 1;
+    string dtype = 2;
+    string delimiter = 3;
+    repeated int32 shape = 4;
+    bool is_sparse = 5;
+    map<string, string> vocabulary = 6;
+    int32 maxID = 7;
+}
+
+message NumericColumn {
+    FieldMeta field_meta = 1;
+}
+
+message BucketColumn {
+    NumericColumn source_column = 1;
+    repeated int32 boundaries = 2;
+}
+
+message CrossColumn {
+    repeated FeatureColumn keys = 1;
+    int32 hash_bucket_size = 2;
+}
+
+message CategoryIDColumn {
+    FieldMeta field_meta = 1;
+    int32 bucket_size = 2;
+}
+
+message SeqCategoryIDColumn {
+    FieldMeta field_meta = 1;
+    int32 bucket_size = 2;
+}
+
+message EmbeddingColumn {
+    oneof category_column {
+        CategoryIDColumn category_col = 1;
+        SeqCategoryIDColumn seq_category_col = 2;
+    }
+    int32 dimension = 3;
+    string combiner = 4;
+    string initializer = 5;
+}
+
+message FeatureColumn {
+    oneof feature_column {
+        NumericColumn nc = 1;
+        BucketColumn bc = 2;
+        CrossColumn cc = 3;
+        CategoryIDColumn catc = 4;
+        SeqCategoryIDColumn seqcatc = 5;
+        EmbeddingColumn embc = 6;
+    }
+}
+
+// TODO(typhoonzero): directly import Session from pkg/server/proto
```

Comment:
I am not sure how to do the direct import. Do we need to include `pkg/server/proto` during `protoc`? If so, is it more convenient to organize all the proto files in one directory?

Reply:
You are probably right, I still need to check the details at https://developers.google.com/protocol-buffers/docs/proto#importing-definitions